mlir/utils/jenkins/premerge-checks.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+import argparse
+import logging
+import os
+import re
+import sys
+import subprocess
+from typing import Tuple, Optional
+import pathspec
+import unidiff
```

Comment:
You may need to `pip3 install` so that those packages are available. If it is already there, feel free to ignore this.